lib/supavisor/handler_helpers.ex

```diff
@@ -126,4 +126,67 @@ defmodule Supavisor.HandlerHelpers do
     :ok = sock_send(sock, msg)
     :ok = sock_close(sock)
   end
+
+  @doc """
+  Takes an allow list of CIDR ranges and filtres them for ranges which contain the address
+  to test.
+
+  If the IP address of the socket is not found an empty list is returned.
+
+  ## Examples
+
+    iex> Supavisor.HandlerHelpers.filter_cidrs(["0.0.0.0/0", "::/0"], {127, 0, 0, 1})
+    ["0.0.0.0/0"]
+
+    iex> Supavisor.HandlerHelpers.filter_cidrs(["71.209.249.38/32"], {71, 209, 249, 39})
+    []
+
+    iex> Supavisor.HandlerHelpers.filter_cidrs(["0.0.0.0/0", "::/0"], {8193, 3512, 34211, 0, 0, 35374, 880, 29492})
+    ["::/0"]
+
+    iex> Supavisor.HandlerHelpers.filter_cidrs(["0.0.0.0/0", "::/0"], :error)
+    []
+
+  """
+
+  @spec filter_cidrs(
+          list(),
+          {byte(), byte(), byte(), byte()}
+          | {char(), char(), char(), char(), char(), char(), char(), char()}
+          | any()
+        ) :: list()
+  def filter_cidrs(allow_list, addr) when is_list(allow_list) and is_tuple(addr) do
+    for range <- allow_list,
+        range |> InetCidr.parse() |> InetCidr.contains?(addr) do
+      range
+    end
+  end
+
+  def filter_cidrs(allow_list, _addr) when is_list(allow_list) do
+    []
+  end
+
+  @spec addr_from_port(port()) ::
+          {:ok,
+           {byte(), byte(), byte(), byte()}
+           | {char(), char(), char(), char(), char(), char(), char(), char()}}
+          | :error
+  def addr_from_port(port) do
+    case :inet.peername(port) do
```

Comment:
This won't work for SSL connections. For a secured connection, you need to call [:ssl.peername/1](https://www.erlang.org/doc/man/ssl#peername-1). Therefore, you need to pass all `Supavisor.sock()` and then choose the right module: `ssl` or `inet`, based on the transport type (`ssl` or `gen_tcp`). Here are some similar functions https://github.com/supabase/supavisor/blob/9186398e5e3ff0e93254a797f2f14dceb207de99/lib/supavisor/handler_helpers.ex#L20-L33

Reply:
Fixed!